templates/docs/examples/patterns/side-navigation/_toggle_script.js

```diff
@@ -0,0 +1,70 @@
+/**
+  Toggles the expanded class on drawer and overlay elements.
+
+  @param {HTMLElement} drawer The side navigation drawer element.
+  @param {Boolean} show Whether to show or hide the drawer.
+*/
+function toggleDrawer(drawer, show) {
+  var overlay = drawer.parentNode.querySelector('.p-side-navigation__overlay');
+
+  if (show) {
+    if (drawer) {
+      drawer.classList.remove('is-collapsed');
+      drawer.classList.add('is-expanded');
+    }
+    if (overlay) {
+      overlay.classList.add('is-expanded');
+    }
+  } else {
+    if (drawer) {
+      drawer.classList.remove('is-expanded');
+      drawer.classList.add('is-collapsed');
+    }
+    if (overlay) {
+      overlay.classList.remove('is-expanded');
+    }
+  }
+}
+
+/**
+  Attaches event listeners for the side navigation toggles
+  @param {HTMLElement} sideNavigation The side navigation element.
+*/
+function setupSideNavigation(sideNavigation) {
+  var toggles = [].slice.call(sideNavigation.querySelectorAll('.js-drawer-toggle'));
+
+  toggles.forEach(function(toggle) {
+    toggle.addEventListener('click', function(event) {
+      event.preventDefault();
+      var drawer = document.getElementById(toggle.getAttribute('aria-controls'));
+
+      if (drawer) {
+        toggleDrawer(drawer, !drawer.classList.contains('is-expanded'));
+      }
+    });
+  });
+}
+
+/**
+  Attaches event listeners for all the side navigations in the document.
+  @param {String} sideNavigationSelector The CSS selector matching side navigation elements.
+*/
+function setupSideNavigations(sideNavigationSelector) {
+  // Setup all side navigations on the page.
+  var sideNavigations = [].slice.call(document.querySelectorAll(sideNavigationSelector));
+
+  sideNavigations.forEach(setupSideNavigation);
+
+  // on resize event remove `is-collapsed` class from drawers to prevent them from animating back in
+  // TODO: this should probably be throttled
+  window.addEventListener('resize', function() {
```

Comment:
@anthonydillon Are you ok with component script listening on resize?

We need that to prevent sidenav animating on screen resize, but it feels like it's getting into area or 'too much JS' for CSS framework.

To use this pattern this (or equivalent) JS is required. And in most cases it will be copied from here blindly.
And it's not bulletproof as it is now. Resize event should be probably throttled (so it's not called million times). This adds more and more to JS that we don't officially publish or support that is required by pattern to work as expected.

Reply:
@anthonydillon Thanks. It was good call.

I changed transition to animation and it allowed to get rid of resize event.
Moving collapsed/expanded to root element also helped with simplifying JS part.